src/pages/homeCollection/printLabels.js

```diff
@@ -52,15 +50,16 @@ const printLabelsTemplate = (name) => {
   document.getElementById("navbarNavAltMarkup").innerHTML = nonUserNavBar(name);
   contentBody.innerHTML = template;
   activeHomeCollectionNavbar();
-
+  if (appState.getState().length === 0) triggerErrorModal('No labels to print');
   generateParticipantCsvGetter(name);
 };
 
 const initializeTotalAddressesToPrint = async () => {
   showAnimation();
   const totalAddresses = await getTotalAddressesToPrint();
-  appState.setState({'totalAddresses': totalAddresses.data})
-  appState.setState({'length': totalAddresses.data.length })
+  const processedAddresses = totalAddresses.data.filter(obj => obj.length !== 0)
+  appState.setState({'totalAddresses': processedAddresses})
+  appState.setState({'length': processedAddresses.length })
```

Comment:
Again, putting data in a separate sub-object inside state will be better. Can be done like this: `appState.setState(state => {labelData: {...state.labelData, totalAddresses: processedAddresses, length: processedAddresses.length})`